zerver/lib/actions.py

```diff
@@ -4863,7 +4869,7 @@ def gather_subscriptions_helper(user_profile: UserProfile,
             stream_dict = {'name': stream['name'],
                            'invite_only': stream['invite_only'],
                            'is_web_public': stream['is_web_public'],
-                           'is_announcement_only': stream['is_announcement_only'],
+                           'stream_post_policy': stream['stream_post_policy'],
```

Comment:
So, what I meant when I suggested preserving this field in the API for mobile apps was that we should include here:

`is_announcement_only: stream['stream_post_policy'] == Stream.STREAM_POST_POLICY_ADMINS`

so that clients like the mobile apps have read access to this field.  The mobile apps don't support creating streams right now, so that should be all we need to do for backwards-compatibility.

(Well, here and in the other dictionaries in `actions.py`)